pennylane/qchem/hamiltonian.py

```diff
@@ -583,24 +573,14 @@ def _molecular_hamiltonian(
         mapping = mapping.strip().lower()
         qubits = len(hf.wires)
 
-        if active_new_opmath():
-            if mapping == "jordan_wigner":
-                h_pl = qml.jordan_wigner(hf, wire_map=wires_map, tol=1.0e-10)
-            elif mapping == "parity":
-                h_pl = qml.parity_transform(hf, qubits, wire_map=wires_map, tol=1.0e-10)
-            elif mapping == "bravyi_kitaev":
-                h_pl = qml.bravyi_kitaev(hf, qubits, wire_map=wires_map, tol=1.0e-10)
-
-            h_pl.simplify()
-        else:
-            if mapping == "jordan_wigner":
-                h_pl = qml.jordan_wigner(hf, ps=True, wire_map=wires_map, tol=1.0e-10)
-            elif mapping == "parity":
-                h_pl = qml.parity_transform(hf, qubits, ps=True, wire_map=wires_map, tol=1.0e-10)
-            elif mapping == "bravyi_kitaev":
-                h_pl = qml.bravyi_kitaev(hf, qubits, ps=True, wire_map=wires_map, tol=1.0e-10)
-
-            h_pl = simplify(h_pl.hamiltonian())
+        if mapping == "jordan_wigner":
+            h_pl = qml.jordan_wigner(hf, wire_map=wires_map, tol=1.0e-10)
+        elif mapping == "parity":
+            h_pl = qml.parity_transform(hf, qubits, wire_map=wires_map, tol=1.0e-10)
+        elif mapping == "bravyi_kitaev":
+            h_pl = qml.bravyi_kitaev(hf, qubits, wire_map=wires_map, tol=1.0e-10)
+
+        h_pl.simplify()
```

Comment:
```suggestion
        h_pl = h_pl.simplify()
```

Otherwise, I'm not sure why we are doing all that expensive work to throw it away.